### PR TITLE
Fix archiver for issue #358

### PR DESCRIPTION
**Pull request by frost242**
This test should cover issue #358

Please wait a bit before applying this PR.

### CODE REVIEW

**Comment by ioguix** [comment]
> Please wait a bit before applying this PR.

yeah, I suppose we need the actual fix to keep thee checks green :)

File: t/01-archiver.t [hunk 1]
```diff
@@ -157,6 +157,44 @@ SKIP: {
     );
 }
 
+# Specific tests for PG9.4+, edge cases with pg_stat_archiver
+# Covers issue #358 and #384
+SKIP: {
+    skip "checking with non superuser role is not supported before v10", 4
+        if $node->version < '9.4';
```
Comment (gleu) [inline t/01-archiver.t]: Should be < '10' according to the message.

**Comment by gleu** [comment]
I took a look at the code, and except the test, it looks fine to me.

**Comment by frost242** [comment]
Thanks for the review.

**Comment by frost242** [comment]
@ioguix do you want to take a look or can I merge this PR ?
